### PR TITLE
More CMake fixes

### PR DESCRIPTION
**Pull request by SoapGentoo**
1. Use `VERSION` specifier in `project()`. This is the modern idiomatic CMake way of specifying the version, for instance see https://rix0r.nl/blog/2015/08/13/cmake-guide/. All regexes for the build scripts have been updated.
2. Namespace all Catch options as suggested by @martinmoene

### CODE REVIEW

File: scripts/releaseCommon.py [hunk 1]
```diff
@@ -128,13 +128,13 @@ def updateConanTestFile(version):
         f.write( line + "\n" )
 
 def updateCmakeFile(version):
-    cmakeParser = re.compile(r'set(CATCH_VERSION_NUMBER \d+\.\d+\.\d+)')
+    cmakeParser = re.compile(r'project(Catch2 LANGUAGES CXX VERSION \d+\.\d+\.\d+)')
```
Comment (horenmar) [inline scripts/releaseCommon.py]: This line can be just deleted completely -- it stopped being used at some point.

**Comment by SoapGentoo** [comment]
@horenmar line removed

**Comment by horenmar** [comment]
Apart from that, `CMake/Findcodecov.cmake` probably shouldn't be changed to make future updates simpler. One possibility is to have `CATCH_ENABLE_COVERAGE` also define `ENABLE_COVERAGE`, the other is to just leave it unprefixed. Personally I'd go with the former.

**Comment by SoapGentoo** [comment]
@horenmar done

**Pull request merged**